accepted/00078-maintenance-windows-web-calendar.md

```diff
@@ -0,0 +1,204 @@
+
+Timezone support is there but has limited functionality. Timezone objects need to be created manually (define the name, the GMT shift and the displayed tooltip yourself) and there is no button to change them on the fly (we would need to implement that).
+
+Ical support is not integrated.
+
+### Integration burden
+
+There are examples to help you get started as well as detailed documentation with use example of the different API elements. Personally I would rank the documentation between `big calendar` and `full calendar`. Overall this project caused me the most trouble to find myself into despite decent docs and examples. No ical support so additional backend work will be required.
+
+## Conclusion
+
+After evaluating each of the three options. My conclusion is that the closest fit for our needs is `Full Calendar`. The ability to read events directly from ical files will spare us from calculating the maintenance schedules in the backend. At the same time it meets all the requirements mentioned above.
+
+After some discussion we came to the conclusion that it would likely be better to not use `Full Calendars` ical API and instead use server side pagination to get the data to display in the calendar, since `Full Calendar` only allows reading data from ical files and is not able to output ical from created events. This does however not affect my decision that `Full Calendar` is the component to use for this implementation.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+## React big calendar
+
+Maintenance windows that occupy more than on day are displayed in the 'all-day' section of the day view and not shown in the time section. This is not really suitable for our use case.
+However there are two potential solutions to that:
+1. Split these maintenance windows into multiple one day events (just for display purpose)
+2. Create a own day view [component](https://github.com/jquense/react-big-calendar/issues/765) based on the existing one
+
+## General risk
+
+Generally switching between calendar components once settled for one won't be straitforward. Whilst the backend part should be salvageable between multiple components the frontend obviously won't. The components presented have complety different APIs, hence switching calendar components will require the frontend part to be reimplemented. 
+
+# Future Improvements
+The current scope includes only the display of maintenance schedules/calendars. In the future we might expand this to support the creation of Maintenance Windows within our webui. All off the options discussed above support the creation of events.
+Reworking the Maintenance Window picker (part of the Action picker if a maintenance window is assigned) will require our struts/jsp action pickers to be migrated to react first and thus will be handled separately.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+- Should the display of the next 10 upcoming MWs be replaced or displayed alongside the calendar?
```

Comment:
Maybe not necessary but, it would be nice to have. If there are many months between the Maintenance Windows. I even thought about adding them in addition the the regular previous and next buttons. But we'll need to see during actual implementation how it work out.